src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpWindowsProxy.cs

```diff
@@ -186,84 +216,127 @@ public void Dispose()
         /// or whitespace separated list, with each entry in the following format:
         /// ([&lt;scheme&gt;=][&lt;scheme&gt;"://"]&lt;server&gt;[":"&lt;port&gt;])
         /// </summary>
-        private static void ParseProxyConfig(string value, out Uri insecureProxy, out Uri secureProxy)
+        private static void ParseProxyConfig(ReadOnlySpan<char> proxyString, out MultiProxy secureProxy, out MultiProxy insecureProxy)
         {
-            insecureProxy = null;
-            secureProxy = null;
-            if (string.IsNullOrEmpty(value))
-            {
-                return;
-            }
+            Uri[] secureUris = new Uri[4], insecureUris = new Uri[4];
+            int secureUriCount = 0, insecureUriCount = 0;
+            int iter = 0;
 
-            if (value.IndexOfAny(s_proxyDelimiters) == -1)
+            while (true)
             {
-                // Optimize out an allocation from Split() when no delimiters are present.
-                ParseProxyConfigPart(value, ref insecureProxy, ref secureProxy);
-                return;
-            }
+                // Skip any delimiters.
+                while(iter < proxyString.Length && Array.IndexOf(s_proxyDelimiters, proxyString[iter]) != -1)
```

Comment:
Nit: space between `while` and `(`.  I really need to get that StyleCop rule enabled :)